include/dgl/heterograph_interface.h

```diff
@@ -0,0 +1,354 @@
+/*!
```

Comment:
@BarclayII @zheng-da Updated the heterograph API. Please have a look.
* HeteroGraphInterface is separated from GraphInterface.
* Most the APIs are similar to GraphInterface but with an extra vtype/etype argument.
* I removed subgraph APIs. It's not clear what kind of semantic should be implemented. On the python side, our `g['node_type']` and `g['stype', 'dtype', 'etype']` syntax is more like a syntax sugar that create a view object only in python.